src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -228,7 +250,7 @@ public ZooKeeperSaslClient getSaslClient() {
      * the public methods will not be exposed as part of the ZooKeeper client
      * API.
      */
-    static class ZKWatchManager implements ClientWatchManager {
+    class ZKWatchManager implements ClientWatchManager {
```

Comment:
Why is this not static any longer? I'd leave the static modifier for this class declaration.